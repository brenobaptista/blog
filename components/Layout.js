import Head from 'next/head'
import Link from 'next/link'
import Image from 'next/image'

import styles from '../styles/Layout.module.css'
import utilStyles from '../styles/Utils.module.css'

const name = 'Breno Baptista'
export const siteTitle = 'Next.js Sample Website'

export default function Layout({ children, home }) {
  return (
    <div className={styles.container}>
      <Head>
        <link rel="icon" href="/favicon.ico" />
        <meta
          name="description"
          content="Learn how to build a personal website using Next.js"
        />
        <meta
          property="og:image"
          content={`https://og-image.now.sh/${encodeURI(
            siteTitle
          )}.png?theme=light&md=0&fontSize=75px&images=https%3A%2F%2Fassets.vercel.com%2Fimage%2Fupload%2Ffront%2Fassets%2Fdesign%2Fnextjs-black-logo.svg`}
        />
        <meta name="og:title" content={siteTitle} />
        <meta name="twitter:card" content="summary_large_image" />
      </Head>
      <header className={styles.header}>
        {home ? (
          <>
            <Image
              src="/images/profile.jpg"
              className={utilStyles.borderCircle}
              alt={name}
              width={128}
              height={128}
            />
            <h1 className={utilStyles.heading2Xl}>{name}</h1>
          </>
        ) : (
          <>
            <Link href="/">
              <a>
                <Image
                  src="/images/profile.jpg"
                  className={utilStyles.borderCircle}
                  alt={name}
                  width={96}
                  height={96}
                />
              </a>
            </Link>
            <h2 className={utilStyles.headingLg}>
              <Link href="/">
                <a className={utilStyles.colorInherit}>{name}</a>
              </Link>
            </h2>
          </>
        )}
      </header>
      <main>{children}</main>
      {!home && (
        <div className={styles.backToHome}>
          <Link href="/">
            <a>‚Üê Back to home</a>
          </Link>
        </div>
      )}
    </div>
  )
}
