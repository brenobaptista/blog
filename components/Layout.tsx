import Head from 'next/head'
import Link from 'next/link'
import Image from 'next/image'

import styles from '../styles/Layout.module.css'
import utilStyles from '../styles/Utils.module.css'

interface Props {
  children: React.ReactNode
  home?: boolean
}

const name = 'Breno Baptista'
export const siteTitle = 'Breno Baptista'

const Layout = ({ children, home }: Props) => (
  <div className={styles.container}>
    <Head>
      <link
        rel="icon"
        href="/favicon.ico"
      />
      <meta
        name="description"
        content="Breno Baptista's Portfolio"
      />
      <meta
        property="og:image"
        content={`https://og-image.now.sh/${encodeURI(
          siteTitle
        )}.png?theme=light&md=0&fontSize=75px&images=https%3A%2F%2Fassets.zeit.co%2Fimage%2Fupload%2Ffront%2Fassets%2Fdesign%2Fnextjs-black-logo.svg`}
      />
      <meta
        name="og:title"
        content={siteTitle}
      />
      <meta
        name="twitter:card"
        content="summary_large_image"
      />
    </Head>
    <header className={styles.header}>
      {home ? (
        <>
          <Image
            src="/images/profile.jpg"
            className={utilStyles.borderCircle}
            alt={name}
            width={128}
            height={128}
          />
          <h1 className={utilStyles.heading2Xl}>
            {name}
          </h1>
        </>
      ) : (
        <>
          <Link href="/">
            <a>
              <Image
                src="/images/profile.jpg"
                className={utilStyles.borderCircle}
                alt={name}
                width={96}
                height={96}
              />
            </a>
          </Link>
          <h2 className={utilStyles.headingLg}>
            <Link href="/">
              <a className={utilStyles.colorInherit}>
                {name}
              </a>
            </Link>
          </h2>
        </>
      )}
    </header>
    <main>
      {children}
    </main>
    {!home && (
      <div className={styles.backToHome}>
        <Link href="/">
          <a>
            ‚Üê Back to home
          </a>
        </Link>
      </div>
    )}
  </div>
)

export default Layout
