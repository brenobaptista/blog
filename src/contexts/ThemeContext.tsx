import { createContext } from 'react'
import { DefaultTheme } from 'styled-components'

import dracula from '@/styles/themes/dracula'

interface IThemeContext {
  theme: DefaultTheme
  toggleTheme: () => void
}

const ThemeContext = createContext<IThemeContext>({
  theme: dracula,
  toggleTheme: () => null
})

export default ThemeContext
