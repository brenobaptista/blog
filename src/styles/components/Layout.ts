import styled from 'styled-components'

export const Container = styled.div`
  max-width: 768px;
  padding: 0 16px;
  margin: 48px auto 96px;

  a {
    color: ${props => props.theme.colors.primary};
  }

  a:hover {
    text-decoration: underline;
  }

  img {
    max-width: 100%;
    display: block;
  }
`

export const Header = styled.header`
  display: flex;
  flex-direction: column;
  align-items: center;

  img {
    border-radius: 50%;
  }
`

export const Separator = styled.hr`
  border: 0;
  height: 1px;
  background: #333;
  background-image: linear-gradient(to right, #ccc, #333, #ccc);
`

export const ShortBio = styled.div`
  display: flex;

  img {
    flex: 1;
    margin-top: 28px;
    border-radius: 50%;
  }

  p {
    flex: 1;
    margin-left: 20px;
    font-style: italic;
  }
`

export const BackToHome = styled.div`
  margin: 48px 0 0;
`
