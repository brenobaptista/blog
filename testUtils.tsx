import { render } from '@testing-library/react'
import { RenderResult } from '@testing-library/react/types'
import { ThemeProvider } from 'styled-components'

import light from './src/styles/themes/light'

const Providers = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
)

const customRender = (ui: React.ReactElement, options = {}): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
