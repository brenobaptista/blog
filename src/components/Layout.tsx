import { ReactNode } from 'react'
import Link from 'next/link'

import Footer from '@/components/Footer'
import Sparkles from '@/components/Sparkles'
import ThemeSwitcher from '@/components/ThemeSwitcher'

import { Container, Toolbar } from '@/styles/components/Layout'
import RetroGlobalStyle from '@/styles/retro-global'

interface ILayout {
  children: ReactNode
}

const Layout = ({ children }: ILayout) => (
  <Container>
    <Toolbar>
      <Link href='/'>Breno Baptista</Link>
      <ThemeSwitcher />
    </Toolbar>
    <main>{children}</main>
    <Footer />
    <Sparkles />
    <RetroGlobalStyle />
  </Container>
)

export default Layout
