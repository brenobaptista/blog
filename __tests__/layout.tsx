import { render } from '../testUtils'
import Layout from '../src/components/Layout'
import ThemeContext from '../src/contexts/ThemeContext'

const renderWithContext = (ui, value) =>
  render(<ThemeContext.Provider value={value}>{ui}</ThemeContext.Provider>)

test('matches snapshot in dark mode', () => {
  const toggleTheme = jest.fn()
  const theme = { mode: 'dark' }
  const { asFragment } = renderWithContext(
    <Layout>
      <></>
    </Layout>,
    { theme, toggleTheme }
  )

  expect(asFragment()).toMatchSnapshot()
})
