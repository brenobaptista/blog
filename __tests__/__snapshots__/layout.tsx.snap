// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot in dark mode 1`] = `
<DocumentFragment>
  .c2 {
  position: absolute;
  height: 0;
  width: 0;
  visibility: hidden;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  margin-bottom: 32px;
  cursor: pointer;
  border-radius: 65px;
  width: 65px;
  height: 32px;
  background: #586693;
}

.c6 {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  border-radius: 28px;
  width: 28px;
  height: 28px;
  background: #f6f5f4;
  box-shadow: 0 0 2px 0 rgba(10,10,10,0.29);
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c1:checked + .c3 .c5 {
  left: calc(100% - 2px);
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c0 {
  max-width: 768px;
  padding: 0 16px;
  margin: 48px auto 96px;
}

.c0 a {
  color: #586693;
}

.c0 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 img {
  max-width: 100%;
  display: block;
}

<div
    class="c0"
  >
    <input
      checked=""
      class="c1 c2"
      id="checkbox"
      type="checkbox"
    />
    <label
      class="c3 c4"
      data-testid="label-checkbox"
      for="checkbox"
      tabindex="0"
    >
      <span
        class="c5 c6"
      />
    </label>
    <main />
  </div>
</DocumentFragment>
`;
